src/main/io/gps.c

```diff
@@ -168,46 +168,51 @@ static const uint8_t ubloxAirborne[] = {
     #endif
 };
 
-// UBlox 6 Protocol documentation - GPS.G6-SW-10018-F
-// SBAS Configuration Settings Desciption, Page 4/210
-// 31.21 CFG-SBAS (0x06 0x16), Page 142/210
-// A.10 SBAS Configuration (UBX-CFG-SBAS), Page 198/210 - GPS.G6-SW-10018-F
-
-#define UBLOX_SBAS_PREFIX_LENGTH 10
+typedef struct {
+    uint8_t preamble1;
+    uint8_t preamble2;
+    uint8_t msg_class;
+    uint8_t msg_id;
+    uint16_t length;
+} ubx_header;
 
-static const uint8_t ubloxSbasPrefix[UBLOX_SBAS_PREFIX_LENGTH] = { 0xB5, 0x62, 0x06, 0x16, 0x08, 0x00, 0x03, 0x07, 0x03, 0x00 };
+typedef struct {
+    uint8_t mode;
+    uint8_t usage;
+    uint8_t maxSBAS;
+    uint8_t scanmode2;
+    uint32_t scanmode1;
+} ubx_sbas;
 
-#define UBLOX_SBAS_MESSAGE_LENGTH 6
-typedef struct ubloxSbas_s {
-    sbasMode_e mode;
-    uint8_t message[UBLOX_SBAS_MESSAGE_LENGTH];
-} ubloxSbas_t;
+typedef union {
+    ubx_sbas sbas;
+} ubx_payload;
 
+typedef struct {
+    ubx_header header;
+    ubx_payload payload;
+} __attribute__((packed)) ubx_message;
 
+typedef union {
+    ubx_message message;
+    uint8_t bytes[58];
```

Comment:
Where does the `58` come from? It's probably better if this is a define to make it self-explanatory and reusable.

Reply:
Well spotted, it's actually something that I forgot to remove when cleaning up. I'm pushing a quick fix, once it's validated I'll squash it.